android/java/org/chromium/chrome/browser/crypto_wallet/fragments/NftGridFragment.java

```diff
@@ -98,29 +107,45 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
     public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
             @Nullable Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.fragment_nft_grid, container, false);
-        mPbAssetDiscovery = view.findViewById(R.id.frag_port_pb_asset_discovery);
+        mPbAssetDiscovery = view.findViewById(R.id.frag_nft_grid_pb_asset_discovery);
         mAddNftsContainer = view.findViewById(R.id.add_nfts_container);
+        mBtnChangeNetwork = view.findViewById(R.id.fragment_nft_grid_btn_change_networks);
+        mBtnChangeNetwork.setOnClickListener(v -> { openNetworkSelection(); });
+        mBtnChangeNetwork.setOnLongClickListener(v -> {
+            if (mNetworkInfo != null) {
+                Toast.makeText(requireContext(), mNetworkInfo.chainName, Toast.LENGTH_SHORT).show();
+            }
+            return true;
+        });
         setUpObservers();
         return view;
     }
 
-    // TODO: refactor this entire method once multi-chain support will be merged.
     private void setUpObservers() {
         if (mWalletModel == null) return;
         mWalletModel.getCryptoModel().getNetworkModel().mCryptoNetworks.observe(
                 getViewLifecycleOwner(),
                 allNetworkInfos -> { mAllNetworkInfos = allNetworkInfos; });
-        mWalletModel.getCryptoModel().getNetworkModel().mDefaultNetwork.observe(
-                getViewLifecycleOwner(), networkInfo -> {
-                    if (networkInfo == null) return;
-                    if (mNetworkInfo != null && !mNetworkInfo.chainId.equals(networkInfo.chainId)) {
+        mNetworkSelectionModel =
+                mWalletModel.getCryptoModel().getNetworkModel().openNetworkSelectorModel(
+                        TAG, NetworkSelectorModel.Mode.LOCAL_NETWORK_FILTER, getLifecycle());
+        mNetworkSelectionModel.getSelectedNetwork().observe(
+                getViewLifecycleOwner(), localNetworkInfo -> {
+                    if (localNetworkInfo == null) return;
+                    if (mNetworkInfo != null
+                            && !mNetworkInfo.chainId.equals(localNetworkInfo.chainId)) {
+                        mAddNftsContainer.setVisibility(View.GONE);
+                        mPbAssetDiscovery.setVisibility(View.VISIBLE);
                         // Clean up list to avoid user clicking on an asset of the previously
                         // selected network after the network has been changed.
                         clearAssets();
                     }
-                    mNetworkInfo = networkInfo;
+                    mNetworkInfo = localNetworkInfo;
+                    mBtnChangeNetwork.setText(
+                            Utils.getShortNameOfNetwork(localNetworkInfo.chainName));
                     updateNftGrid();
                 });
+
         mPortfolioModel.mNftModels.observe(getViewLifecycleOwner(), nftDataModels -> {
             if (mPortfolioModel.mPortfolioHelper == null) return;
```

Comment:
Needs a null check?
[Prefer to provide default values than adding null checks](https://github.com/brave/brave-core/pull/17967#discussion_r1165227872)

Reply:
There is no crash because of using the domain layer with live data.